### ISSUE TITLE
Macros are being expanded when emitting suggestions

### ISSUE THREAD

**Issue by trixnz** [rust-lang-nursery/rust-clippy#1148]
When clippy reports a suggestion with a macro in the code, it seems to print the expanded form. Using the example from README.md, you can see it is not behaving as expected: 

``` rust
fn main(){                             
    let x = Some(1u8);                 
    match x {                          
        Some(y) => println!("{:?}", y),
        _ => ()                        
    }                                  
}
```

results in:

```
warning: you seem to be trying to use match for destructuring a single pattern. Consider using `if let`, #[warn(single_match)] on by default
 --> src\main.rs:3:5
  |
3 |     match x {
  |     ^
  |
help: try this
  |     if let Some(y) = x { $ crate :: io :: _print ( format_args ! ( $ ( $ arg ) * ) ) }
  = help: for further information visit https://github.com/Manishearth/rust-clippy/wiki#single_match
```

**Comment by Manishearth** [comment]
This seems to be `snippet()` misbehaving, cc @jonathandturner

**Comment by Manishearth** [comment]
er, `span_to_snippet`

**Comment by trixnz** [comment]
After doing a bit more research, this only seems to affect match expressions without scope arms. This produces the desired output:
```rust
fn main() {
    let x = Some(1u8);
    match x {
        Some(y) => {
            println!("{:?}", y);
        }
        _ => (),
    }
}
```
results in:
```
warning: you seem to be trying to use match for destructuring a single pattern. Consider using `if let`, #[warn(single_match)] on by default
 --> src\main.rs:3:5
  |
3 |     match x {
  |     ^
  |
help: try this
  |     if let Some(y) = x {
  |     println!("{:?}", y);
  | }
  = help: for further information visit https://github.com/Manishearth/rust-clippy/wiki#single_match
```

**Comment by jonathandturner** [comment]
Agreed - looks like we're (I'm) not sorting the help with a snippet to be after the ones without snippets.  Filed https://github.com/rust-lang/rust/issues/35460

**Comment by oli-obk** [comment]
@jonathandturner the issue here is not the snippet ordering, but the expansion of the `println` macro
